Quadriga/src/main/java/edu/asu/spring/quadriga/dao/impl/UserDAO.java

```diff
@@ -389,4 +391,33 @@ public QuadrigaUserDeniedDTO getDeniedUser(String id) {
         return (QuadrigaUserDeniedDTO) sessionFactory.getCurrentSession().get(QuadrigaUserDeniedDTO.class, id);
     }
     
+    
+    public QuadrigaUserDTO findUserByProviderUserId(String userId, String provider){
+        QuadrigaUserDTO userDTO  = null;
+      
+            Query query = sessionFactory.getCurrentSession().createQuery(
+                    " from QuadrigaUserDTO user where user.userIdOfProvider=:userId AND user.provider=:provider");
+            query.setParameter("userId", userId);
+            query.setParameter("provider", provider);
+            List resultList = query.list();
+            
+            if(resultList != null){
+                if(resultList.size() > 0){ 
+                    userDTO = (QuadrigaUserDTO)(resultList.get(0));
```

Comment:
Yes. Fixed to return the value directly instead of assigning to a variable first before returning.